contracts/Account.sol

```diff
@@ -67,6 +66,44 @@ contract Account is IAccount, Ownable, SafetyWithdraw {
         transferOwnership(governance);
     }
 
+    /**
+     * @notice Adjust the max leverage as insurance pool slides from 100% of target to 0% of target
+     */
+    function realMaxLeverage(address market) public view override returns(int256) {
+        ITracerPerpetualSwaps _tracer = ITracerPerpetualSwaps(market);
+        IInsurance insurance = IInsurance(insuranceContract);
+        int256 baseMaxLeverage = _tracer.maxLeverage();
+
+        int256 percentFull =
+            insurance.getPoolHoldings(market).toInt256()
+            .mul(PERCENT_PRECISION)
+            .div(insurance.getPoolTarget(market).toInt256()).mul(100);
+        int256 cliff = _tracer.INSURANCE_DELEVERAGING_CLIFF();
+
+        if (percentFull > cliff.mul(PERCENT_PRECISION)) {
+            return baseMaxLeverage;
+        }
+
+        // Use int256 for compatibility with baseMaxLeverage
+        //int256 oneXLeverage = 10000;
+        // Linear function intercepting points (1, 0) and (INSURANCE_DELEVERAGING_CLIFF, baseMaxLeverage)
```

Comment:
does this imply at percentFull = 1%, the real max leverage is 0?.... so nobody can take any positions if the pool is less than 1% full?

Reply:
It's the opposite. At 0% full insurance pool, max leverage is 1